src/combinations.rs

```diff
@@ -126,3 +126,17 @@ impl<I> FusedIterator for Combinations<I>
     where I: Iterator,
           I::Item: Clone
 {}
+
+pub(crate) fn checked_binomial(mut n: usize, mut k: usize) -> Option<usize> {
+    if n < k {
+        return Some(0);
+    }
+    // `factorial(n) / factorial(n - k) / factorial(k)` but trying to avoid it overflows:
+    k = (n - k).min(k); // symmetry
+    let mut c = 1;
+    for i in 1..=k {
+        c = (c / i).checked_mul(n)?.checked_add((c % i).checked_mul(n)? / i)?;
+        n -= 1;
+    }
+    Some(c)
+}
```

Comment:
Could we at least exhausively test this implementation? (I.e. generate a lookup-table and check the results?)

Or even implement this via a lookup table? (Not urgent imho.)

Reply:
Yes, I just was not sure how. I mean that I don't think we want it public so not in the tests folder. So a doctest or a `#[test]` function which is not a thing we can find in the entire library, maybe there is a reason.

Reply:
Maybe a hardwired two-dimensional array of `u128` with a reference from where we took it, and then simply iterate over all pairs of supported `k`/`n` and assert that the `usize::try_from(lookuptable_holding_u128[n][k])` equals the result from the function?

Reply:
Sure. Something like this below inside `combinations.rs`?

    #[test]
    fn test_checked_binomial() {
        let table: [[u128; N]; K] = [[1, ...], ...]; // N K constants.
        for n in 0..N {
            for k in 0..K {
                assert_eq!(checked_binomial(n, k), usize::try_from(lookuptable_holding_u128[n][k]).ok());
            }
        }
    }

Reply:
Runs fast for a limit of 100 (0.01 second I think, 65% of the last row overflow). I made it run on 10000 to see though (20-30 seconds, and like 98% overflow).
Maybe you have another meaning of exhaustive testing than me but for small k, it won't overflow for a long long time (`6_074_001_001` would be the smaller `n` for which `(0..n).combinations(2).count()==n*(n-1)/2` would overflow for a fixed `k=2`). So we have to have a limit on `n`. 100 might seem small but already 65% of the row has already overflowed.